sections/header.liquid

```diff
@@ -738,6 +738,7 @@
     hide() {
       if (this.headerIsAlwaysSticky) return;
       this.header.classList.add('shopify-section-header-hidden', 'shopify-section-header-sticky');
+      document.documentElement.style.setProperty('--header-height', `${this.header.offsetHeight}px`);
```

Comment:
Would it make sense to move this into `connectedCallback()` ? This way it's setting the variable to the right value without waiting for `hide()` to run 🤔